config/kube.py

```diff
@@ -0,0 +1,337 @@
+            for label in labels:
+                _label_node(label)
+
+    except requests.exceptions.RequestException as e:
+        clicommon.do_exit("Download failed: {}".format(str(e)))
+
+    except OSError as e:
+        clicommon.do_exit("Download failed: {}".format(str(e)))
+
+    _troubleshoot_tips()
+
+
+def kube_reset():
+    lock_fd = _take_lock()
+    if not lock_fd:
+        log.log_error("Lock {} is active; Bail out".format(LOCK_FILE))
+        return
+
+    # Remove a key label and drain/delete self from cluster
+    # If not, the next join would fail
+    #
+    if os.path.exists(KUBE_ADMIN_CONF):
+        _label_node("enable_pods-")
+        clicommon.run_command(
+                "kubectl --kubeconfig {} --request-timeout 20s drain {} --ignore-daemonsets".format(
+                    KUBE_ADMIN_CONF, device_info.get_hostname()),
+                ignore_error=True)
+        clicommon.run_command(
+                "kubectl --kubeconfig {} --request-timeout 20s delete node {}".format(
+                    KUBE_ADMIN_CONF, device_info.get_hostname()),
+                ignore_error=True)
+
+    clicommon.run_command("kubeadm reset -f", ignore_error=True)
+    clicommon.run_command("rm -rf /etc/cni/net.d")
+    clicommon.run_command("rm -f {}".format(KUBE_ADMIN_CONF))
+    clicommon.run_command("systemctl stop kubelet")
+    clicommon.run_command("systemctl disable kubelet")
+
+
+def kube_join(force=False):
+    lock_fd = _take_lock()
+    if not lock_fd:
+        log.log_error("Lock {} is active; Bail out".format(LOCK_FILE))
+        return
+
+    db_data = Db().get_data('KUBERNETES_MASTER', 'SERVER')
+    if not db_data or 'IP' not in db_data or not db_data['IP']:
+        log.log_error("Kubernetes server is not configured")
+
+    if db_data['disable'].lower() != "false":
+        log.log_error("kube join skipped as kubernetes server is marked disabled")
+        return
+
+    if not force:
+        if _is_connected(db_data['IP']):
+            # Already connected. No-Op
+            return
+
+    kube_reset()
+    _do_join(db_data['IP'], db_data['insecure'])
+
+
+@click.group(cls=clicommon.AbbreviationGroup)
+def kubernetes():
+    """kubernetes command line"""
+    pass
+
+
+# cmd kubernetes join [-f/--force]
+@click.command()
+@click.option('-f', '--force', help='Force a join', is_flag=True)
+def join(force):
+    kube_join(force=force)
+
+kubernetes.add_command(join)
+
+
+# cmd kubernetes reset
+@click.command()
+def reset():
+    kube_reset()
+
+kubernetes.add_command(reset)
+
+
+# cmd kubernetes server
+@click.group()
+def server():
+    """ Server configuration """
+    pass
+
+kubernetes.add_command(server)
+
+
+# cmd kubernetes server IP
+@click.command()
+@click.argument('vip')
+def ip(vip):
+    """Specify a kubernetes cluster VIP"""
+    if not netaddr.IPAddress(vip):
+        click.echo('Invalid IP address %s' % vip)
+        return
+    _update_kube_server('IP', vip)
+
+server.add_command(ip)
```

Comment:
Rather than calling `kubernetes.add_command(x)`, you should simply be able to change the decorator from:

```python
@click.command()
```

to:

```python
@server.command()
```

Reply:
done